ui/client/containers/CustomTabs.tsx

```diff
@@ -0,0 +1,56 @@
+import PropTypes, {Validator} from "prop-types"
+import React from "react"
+import {connect} from "react-redux"
+
+export class CustomTabs extends React.Component<{ settings: $TodoType[], match: $TodoType }> {
+
+  static path: string
+  static header: string
+  static propTypes: {
+    settings: Validator<NonNullable<any[]>>,
+    match: Validator<NonNullable<object>>,
+  }
+
+  constructor(props) {
+    super(props)
+  }
+
+  render() {
+    const {match: {params: {id}}}  = this.props
+    const ref = "customTabsFrame"
+    const tab = this.props.settings
+      .find(o => o.id == id)
+
+    if (tab) {
+      return (
+        <div className="Page">
+          <iframe
+            ref={ref}
+            src={tab.url}
+            width="100%"
+            height={window.innerHeight}
+            frameBorder="0"
+          />
+        </div>
+      )
+    }
+
+    return (<div/>)
```

Comment:
I wonder if 404 wouldn't be nicer in this case? Or at least let's add } else { :)